packages/lib/modules/lbp/steps/sale-structure/WeightsChart.tsx

```diff
@@ -0,0 +1,169 @@
+import { addHours, differenceInDays } from 'date-fns'
+import ReactECharts, { EChartsOption } from 'echarts-for-react'
+import * as echarts from 'echarts/core'
+import { bn } from '@repo/lib/shared/utils/numbers'
+
+export function WeightsChart({
+  startWeight,
+  endWeight,
+  startDate,
+  endDate,
+}: {
+  startWeight: number
+  endWeight: number
+  startDate: Date
+  endDate: Date
+}) {
+  const launchTokenData = interpolateData(startWeight, endWeight, startDate, endDate)
+  const collateralTokenData = invertData(launchTokenData)
+
+  const chartInfo: EChartsOption = {
+    xAxis: {
+      show: true,
+      type: 'value',
+      axisLine: { show: false },
+      splitLine: { show: false },
+      axisTick: { show: false },
+      min: startDate.getTime(),
+      max: endDate.getTime(),
+      interval: 24 * 60 * 60 * 1000,
```

Comment:
maybe use `hoursToMilliseconds` from `date-fns` instead of these magic numbers?

Reply:
changed